server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -260,23 +260,97 @@ class ArrivalsGraphStage(name: String = "",
       mergeBestFieldsFromSources(baseArrival, mergeArrival(key).getOrElse(baseArrival))
     }
 
+
+    def arrivalMapForSourceType(searchSourceType: ArrivalsSourceType): SortedMap[UniqueArrival, Arrival] = searchSourceType match {
+      case LiveArrivals => liveArrivals
+      case LiveBaseArrivals => liveBaseArrivals
+      case BaseArrivals => forecastBaseArrivals
+    }
+
+    def potentialKeys(mergeKey: UniqueArrival, searchArrivalMap: SortedMap[UniqueArrival, Arrival]): Iterable[UniqueArrival] =
+      searchArrivalMap.keys.filter(_.potentialKey(mergeKey, 1 * 60 * 60 * 1000))
+
+    def mergeScheduledApproximation(mergeKey: UniqueArrival, mergeKeySourceType: ArrivalsSourceType, foundArrival: Arrival, searchSourceType: ArrivalsSourceType) = {
```

Comment:
Looking at this again, I think `mergeKeySourceType` should probably be `arrivalSourceType` since it's related to the arrival rather than its key.

Reply:
It is just different way of looking at code . I can change it if it make more sense to you  .  All are arrival , in itself arrival doesn't help to understand the code . where as mergeKeySourceType explain that the key which we calling to merge comes from SourceType where it found .

Reply:
The arrival is from a certain source, not the key. The key can be used to look up arrivals from any source, but it comes from the arrival itself

Reply:
Let's @bnppl opinion here as I think mergeKeySourceType make sense to me . But if he understand it the way you saying it then I can change it . Sometime it better to get 3rd perspective .

Reply:
This is quite a fuzzy logic where we trying to merge key from 2 different source type . I can understand if more thought is given to it .